drivers/flash/flash_stm32_xspi.c

```diff
@@ -1196,7 +1389,9 @@ static int flash_stm32_xspi_write(const struct device *dev, off_t addr,
 			break;
 		}
 	}
+	goto write_end;
```

Comment:
goto not required here, but required after stm32_xspi_mem_ready.

Reply:
agree, 
moved to line 1347
```
	if (ret != 0) {
		LOG_ERR("XSPI: write not ready");
		goto write_end;
	}
```
that was just keeping the same structure as flash_stm32_ospi driver